software_checklist.md

```diff
@@ -0,0 +1,147 @@
+- If there are possible issues reported by such a tool, have they all been either fixed or understood to not be important?
+
+> - [x] Sufficiently addressed
+> - [ ] Some work remaining or caveats
+> - [ ] Needs to be addressed
+>
+> We use [pylint, automated by codacy](README.md#static-analysis), to check code going into the repository. We make some exceptions to the pylint defaults – these are [specified in the repo](.pylintrc). New and altered code is checked and issues dealt with prior to being integrated into the `master` branch.
+
+### Is the code clean, generally understandable and readable and written according to good software engineering principles?
+
+- Is it modular?  Are the internal implementation details of one module hidden from other modules?
+- Commented where necessary?
+- Avoiding red flags such as very long functions, global variables, copy and pasted code, etc.?
+
+> - [x] Sufficiently addressed
+> - [ ] Some work remaining or caveats
+> - [ ] Needs to be addressed
+>
+> Code is modular, documented, low in duplication and well laid out.
+
+### Is there sufficient documentation?
+
+- Is there a readme?
+- Does the code have user documentation?
+- Does the code have developer documentation?
+- Does the code have algorithm documentation? e.g. something that describes how the model is actually simulated, or inference is performed?
+- Is all the documentation up to date?
+
+> - [x] Sufficiently addressed
+> - [ ] Some work remaining or caveats
+> - [ ] Needs to be addressed
+>
+> We have [user](README.md), [developer](https://readthedocs.org/projects/simple-network-sim/badge/?version=latest) and [algorithm](doc/model_documentation/model_description.pdf) documentation. A GitHub issue has been raised to address the areas of documentation that need to be updated.
+
+### Is there suitable collaboration infrastructure?
+
+- Is the code in a version-controlled repository?
+- Is there a license?
+- Is an issue tracker used?
+- Are there contribution guidelines?
+
+> - [x] Sufficiently addressed
+> - [ ] Some work remaining or caveats
+> - [ ] Needs to be addressed
+>
+> We use a [GitHub repository](https://github.com/ScottishCovidResponse/simple_network_sim), with a [BSD license](LICENSE.txt), issue tracking and [contribution guidelines](contributing.md).
+
+### Are software dependencies listed and of appropriate quality?
+
+> - [ ] Sufficiently addressed
+> - [ ] Some work remaining or caveats
+> - [x] Needs to be addressed
+>
+> Dependencies are specified in a conda [spec-file](spec-file.txt). We depend on the SCRC Data Pipeline and [SCRC Data Pipeline API](https://github.com/ScottishCovidResponse/data_pipeline_api) which are not yet sufficiently developed to ensure robust, reproducible operation - they are under intensive development and we expect to have this soon.
```

Comment:
What is missing from the data pipeline api point of view? We are able to pull the dependency from conda.

Reply:
My point is that there are issues with `data_pipeline_api` dependency in itself that still need work (e.g. docs, test coverage, this kind of checking) - we're handling the dependency fine, the dependency itself (I think we all agree) needs more time spent on it. Does this make sense?

Reply:
I see, I thought this checklist was only for simple_network_sim and that the data pipeline api would have its own checklist, but that makes sense.

Reply:
Ah, you may mean that we are still lacking the scripts to upload and download things, right? That makese sense. I thought it would be just a data_pipeline_api command, but from your end-to-end PR it seems to be a bit more than that.